src/client.rs

```diff
@@ -0,0 +1,152 @@
+        &self.hopr
+    }
+}
+
+impl Edgli {
+    pub async fn new(
+        cfg: HoprLibConfig,
+        db_data_path: &Path,
+        hopr_keys: HoprKeys,
+    ) -> anyhow::Result<Self> {
+        if let hopr_lib::config::HostType::IPv4(address) = &cfg.host.address {
+            let ipv4: std::net::Ipv4Addr = std::net::Ipv4Addr::from_str(address)
+                .map_err(|e| EdgliError::ConfigError(e.to_string()))?;
+
+            if ipv4.is_loopback() && !cfg.protocol.transport.prefer_local_addresses {
+                Err(hopr_lib::errors::HoprLibError::GeneralError(
+                    "Cannot announce a loopback address".into(),
+                ))?;
+            }
+        }
+
+        info!(
+            packet_key = hopr_lib::Keypair::public(&hopr_keys.packet_key).to_peerid_str(),
+            blockchain_address = hopr_lib::Keypair::public(&hopr_keys.chain_key)
+                .to_address()
+                .to_hex(),
+            "Node public identifiers"
+        );
+
+        // TODO: stored tickets need to be emitted from the Hopr object (addressed in #7575)
+        //
+        // edge_clients do not store tickets, since they are originators only.
+        let node_db = init_hopr_node_db(
+            db_data_path
+                .to_str()
+                .ok_or_else(|| EdgliError::ConfigError("Invalid database path".into()))?,
+            true,
+            false,
+        )
+        .await?;
+
+        #[cfg(feature = "blokli")]
+        let chain_connector = Arc::new(
+            init_blokli_connector(
+                &hopr_keys.chain_key,
+                None, // read the provider URL from the default env variable for now
+                cfg.safe_module.module_address,
+            )
+            .await?,
+        );
+
+        // Create the node instance
+        info!("Creating the HOPR edge node instance from hopr-lib");
+        let node = Arc::new(
+            hopr_lib::Hopr::new(
+                (&hopr_keys.chain_key, &hopr_keys.packet_key),
+                #[cfg(feature = "blokli")]
+                chain_connector.clone(),
+                node_db,
+                cfg.clone(),
+            )
+            .await?,
+        );
+
+        node.run(hopr_ct_telemetry::ImmediateNeighborProber::new(
+            Default::default(),
+        ))
+        .await?;
+
+        Ok(Self {
+            hopr: node,
+            #[cfg(feature = "blokli")]
+            blokli_connector: chain_connector,
+        })
+    }
+
+    pub fn as_hopr(&self) -> Arc<HoprEdgeClient> {
+        self.hopr.clone()
+    }
+
+    /// Run a node with HOPR edge strategies integrated.
+    ///
+    /// Edge strategies comprise:
+    /// 1. automatically funding the channels when out of funds
+    /// 2. automatically closing the channels in pending to close state
+    #[cfg(feature = "blokli")]
+    pub fn run_reactor_from_cfg(
+        &self,
+        cfg: super::strategy::MultiStrategyConfig,
+    ) -> anyhow::Result<AbortHandle> {
```

Comment:
If you relax the Clone requirement on the Edgli object, you can store the AbortHandle inside the instance, so it gets killed when the instance is dropped.